x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtUtilTests.java

```diff
@@ -6,49 +6,158 @@
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("http://example.com:8080/jwkset.json"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("http://example.com/path/"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("http://example.com:80/path/"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("http://example.com:8080/path/"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("http://example.com/path/jwkset.json"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("http://example.com:80/path/jwkset.json"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("http://example.com:8080/path/jwkset.json"));
     }
 
-    // All JWSAlgorithm values in Family.HMAC_SHA, Family.RSA, and Family.EC (except ES256K, handled separately)
-    public void testSignedJwtGenerateSignVerify() throws Exception {
-        for (final JWSAlgorithm signatureAlgorithm : JwtUtil.SUPPORTED_JWS_ALGORITHMS) {
-            assertThat(this.helpTestSignatureAlgorithm(signatureAlgorithm), is(true));
-        }
+    public void testParseHttpsUriProblemsRejected() {
+        // Reject invalid HTTPS URIs where parse failed
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https:"));
+        // Reject valid HTTPS URIs where host is missing
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https:/"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://:443"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://:443/"));
+        // Reject valid HTTPS URIs where file path is missing
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://example.com"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://example.com/"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://example.com:443"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://example.com:443/"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://example.com:8443"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://example.com:8443/"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://example.com/path/"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://example.com:443/path/"));
+        expectThrows(SettingsException.class, () -> JwtUtil.parseHttpsUri("https://example.com:8443/path/"));
```

Comment:
This is wrong ,I think we need to rework our validation here. There is no need to expect that JWKSet URLs will point to a specific file. This is not mandated anywhere. For instance, when using Google as an OP, the JWKs are available at `https://www.googleapis.com/oauth2/v3/certs`